src/privileged/FirefoxMonitor.jsm

```diff
@@ -170,45 +175,34 @@ this.FirefoxMonitor = {
     this.strings = Services.strings.createBundle(`data:text/plain;base64,${b64}`);
   },
 
-  _loadBreachesTimer: null,
-  _breachesLastModified: 0,
+  kRemoteSettingsKey: "fxmonitor-breaches",
   async loadBreaches() {
-    let response;
-    try {
-      response = await fetch(this.breachListURL, {
-        credentials: "omit",
-        headers: {
-          "If-Modified-Since": this._breachesLastModified,
-        },
+    const { RemoteSettings } = ChromeUtils.import("resource://services-settings/remote-settings.js", {});
+
+    let populateSites = (data) => {
+      this.domainMap.clear();
+      data.forEach(site => {
+        this.domainMap.set(site.Domain, {
+          Name: site.Name,
+          PwnCount: site.PwnCount,
+          Year: (new Date(site.BreachDate)).getFullYear(),
+        });
       });
-    } catch (e) {
-      // Fetch only rejects on network failures or other anomalies;
-      // response will be undefined and we'll return early.
     }
 
-    // Arm the refresh timer already, since we may return early if we 304'd.
-    // Commented out for now since we are packaging breaches in the addon.
-    // this._loadBreachesTimer = setTimeout(() => this.loadBreaches(), this.breachRefreshTimeout);
+    RemoteSettings(this.kRemoteSettingsKey).on("sync", (event) => {
+      const { data: { current } } = event;
+      populateSites(current);
+    });
 
-    // If the list hasn't been updated since we last checked, the server
-    // will send a 304 response. In any case, we don't handle anything
-    // except a 200 OK.
-    if (!response || response.status !== 200) {
-      return;
+    const data = await RemoteSettings(this.kRemoteSettingsKey).get();
+    if (data && data.length) {
+      populateSites(data);
```

Comment:
What are the different context in which these different calls to `populateSites` would happen? I.e., why are we calling it during `RemoteSettings.on("sync")` *AND* after `RemoteSettings().get()` ?

Could this result in a needless double-call?